test/NFTOpt.tests.ts

```diff
@@ -176,6 +201,115 @@ describe("NFTOpt Tests", function () {
         it("should test that method can be called", async function () {
             expect(NFTOptCTR.connect(buyer).createOption(0)).to.not.throw;
         });
+
+        it("should fail when the option with the specified id does not exist", async function () {
+            await expect(
+                NFTOptCTR.connect(seller).createOption(0)
+            ).to.be.revertedWith("Option with the specified id does not exist");
+        });
+
+        it("should fail when the option is already fulfilled by a seller", async function () {
+            await publishDummyOptionRequest();
+
+            await expect(
+                NFTOptCTR.connect(seller).createOption(1, {value: dummyOptionRequest.strikePrice})
+            ).to.not.reverted;
+
+            await expect(
+                NFTOptCTR.connect(seller).createOption(1, {value: dummyOptionRequest.strikePrice})
+            ).to.be.revertedWith("Option is already fulfilled by a seller");
+        });
+
+        it("should fail when the option is not in the request state", async function () {
+            await publishDummyOptionRequest();
```

Comment:
i think don't need await here, `publishDummyOptionRequest` already waits for its call to complete